File: Python/palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = ''.join(c for c in s.lower() if c.isalnum())
        length = len(s)

        if length < 2:
            return True
        
        for i in range(0, length//2):
            if s[i] != s[length-i-1]:
                return False
            
        return True




# Create an instance of the Solution class
solution = Solution()

# Test String
test_cases = [
    "A man, a plan, a canal: Panama",
    "race car",
    "not a palindrome",
    # add more test cases here
]

for test_case in test_cases:
    result = solution.isPalindrome(test_case)
    print(f'The string "{test_case}" is {"a palindrome" if result else "not a palindrome"}')

File: Java/palindrome.java
public class palindrome {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase().replaceAll("[^a-z0-9]", "");
        int len = s.length();
        if(len<2){
            return true;
        }
        for(int i = 0; i < len/2; i++){
            if(s.charAt(i) != s.charAt(len-i-1)){
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        palindrome solution = new palindrome();
        String[] testCases = {
            "A man, a plan, a canal: Panama",
            "race car",
            "not a palindrome",
            // add more test cases here
        };
        for (String testCase : testCases) {
            boolean result = solution.isPalindrome(testCase);
            System.out.println("The string \"" + testCase + "\" is " + (result ? "" : "not ") + "a palindrome.");
        }
    }
}